var/spack/repos/builtin/packages/model-traits/package.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class ModelTraits(CMakePackage):
+    """ Model setup and querying in C++. """
+
+    homepage = "https://github.com/jacobmerson/model-traits/"
+    url      = "https://github.com/jacobmerson/model-traits/archive/refs/tags/v0.1.0.tar.gz"
+    git      = "https://github.com/jacobmerson/model-traits.git"
+
+    maintainers = ['jacobmerson']
+
+    version('0.1.1', sha256='75af53b4f576071570fdcfa7a4ce150b935cf21368da41d16d8377c5b3b93713')
+    version('0.1.0', sha256='ff7c1c5be6977f1d3dc592e8b6c5bff5a8b7ea80d0f059d85c02300bdb8faf2c')
+    version('main', branch='main')
+
+    variant('yaml', default=True, description='build the Yaml IO backend')
+    variant('simmetrix', default=False, description='build the Simmetrix backend')
+    variant('pumi', default=False, description='build the pumi examples')
+
+    depends_on('yaml-cpp@0.6.3:', when='+yaml')
+    depends_on('catch2@3.0.0-preview3:', type='test')
+    depends_on('pumi', when='+pumi')
+    depends_on('simmetrix-simmodsuite', when='+simmetrix')
+    depends_on('fmt@7.1.3')
+    depends_on('cmake@3.14.0:', type='build')
+    depends_on('mpi', when='+simmetrix')
+    depends_on('mpi', when='+pumi')
+    depends_on('cmake@3.14:', type='build')
+
+    def cmake_args(self):
+        try:
+            self.compiler.cxx14_flag
+        except UnsupportedCompilerFlag:
+            InstallError('model-traits requires a C++14-compliant C++ compiler')
```

Comment:
Wouldn't this be checked already by CMake? Referring to the use of [this](https://cmake.org/cmake/help/v3.14/prop_tgt/CXX_STANDARD.html) or similar CMake properties.

Reply:
This is checked by CMake like you say, but putting the check in the package.py makes things fail faster. But, having this redundancy would make it easier for things to get out of sync (and thinking about it this is probably a work around for build systems which don't check the language support).

I think I will remove it.